### ISSUE TITLE
Bug: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

### ISSUE THREAD

**Issue by jomarquez21** [typescript-eslint/typescript-eslint#7087]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

```typescript
no code.
```

### ESLint Config

```javascript
module.exports = {
  env: {
    browser: true,
    es2021: true,
    node: true,
    jest: true,
    'jest/globals': true,
  },
  root: true,
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:@typescript-eslint/recommended',
    'airbnb-base',
    'airbnb-typescript/base',
    'plugin:import/errors',
    'plugin:import/warnings',
    'plugin:jsdoc/recommended-error',
    'plugin:promise/recommended',
    'plugin:prettier/recommended',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 'latest',
    sourceType: 'module',
    project: './tsconfig.json',
  },
  plugins: [
    '@typescript-eslint',
    'import',
    'filenames',
    'header',
    'jsdoc',
    'deprecation',
    'promise',
    'prettier',
    'sort-class-members',
  ],
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "module": "commonjs",
    "esModuleInterop": true,
    "target": "es2019",
    "noImplicitAny": true,
    "moduleResolution": "node",
    "sourceMap": true,
    "outDir": "./dist",
    "rootDir": "./",
    "baseUrl": ".",
    "allowJs": true,
    "checkJs": true,
    "noEmit": false,
    "lib": ["esnext"],
    "paths": {},
    "typeRoots": ["node_modules/@types", "src/types"],
    "forceConsistentCasingInFileNames": true,
    "listEmittedFiles": false,
    "listFiles": false,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "traceResolution": false,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "strict": true,
    "strictPropertyInitialization": false,
    "noFallthroughCasesInSwitch": true,
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "pretty": true
  },
  "exclude": ["node_modules", "dist"],
  "compileOnSave": false
}
```

### Expected Result

No warning

### Actual Result

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.1.0

YOUR TYPESCRIPT VERSION: 5.1.3

Please only submit bug reports when using the officially supported version.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^5.59.8`
|  `@typescript-eslint/parser` | `^5.59.8`
| `TypeScript`                           | `5.1.3` |
| `ESLint`                               | `8.42.0` |
| `node`                                 | `18.16.0` |

**Comment by Josh-Cena** [comment]
Please see the pinned issue https://github.com/typescript-eslint/typescript-eslint/issues/6934
